packages/java/endpoint/src/main/java/com/vaadin/hilla/route/records/ClientViewConfig.java

```diff
@@ -42,25 +50,103 @@ public void add(String key, Object value) {
 
     /**
      * Get all unknown values.
+     *
      * @return a map of unknown values
      */
     @JsonAnyGetter
     public Map<String, Object> getOther() {
         return other;
     }
 
+    public String getTitle() {
+        return title;
+    }
+
+    public String[] getRolesAllowed() {
+        return rolesAllowed;
+    }
+
+    public boolean isRequiresLogin() {
+        return requiresLogin;
+    }
+
+    public String getRoute() {
+        return route;
+    }
+
+    public boolean isLazy() {
+        return lazy;
+    }
+
+    public boolean isRegister() {
```

Comment:
Same as previous comment, `register` is not informative enough about what is it exactly doing/affecting, if it's enabling the users to override the default automatic registration of views, I'd suggest renaming it to `autoRegistered` (or `automaticallyRegistered` to be more verbose) which leads to having something like `isAutoRegistered` instead of `isRegister` as the accessor.